cli/flox/src/commands/search.rs

```diff
@@ -75,12 +73,22 @@ impl Search {
 
         let results = if let Some(client) = flox.catalog_client {
             tracing::debug!("using catalog client for search");
+            let parsed_search = match parse_search_term(&self.search_term) {
+                ParsedSearch::Clean(term) => term,
+                ParsedSearch::VersionStripped(term) => {
+                    message::warning(formatdoc! {"
+                        'flox search' ignores version specifiers.
+                        To see available versions of a package, use 'flox show'
```

Comment:
The suggested warning text in #1506 does duplicate the `flox show` suggested but I think that makes sense to contextualise the warning cc @jennymahmoudi @ghudgins 
```
% flox search hello@1.2.3
⚠️  'flox search' ignores version specifiers.
To see available versions of a package, use 'flox show'

hello                         A program that produces a familiar, friendly greeting
hello-unfree                  An example package with unfree license (for testing)
texlivePackages.othello       Modification of a Go package to create othello boards
javaPackages.mavenHello_1_1   Maven Hello World
javaPackages.mavenHello_1_0   Maven Hello World
texlivePackages.othelloboard  Typeset Othello (Reversi) diagrams of any size, with annotations
gnome.iagno                   Computer version of the game Reversi, more popularly called Othello

Use 'flox show <package>' to see available versions
```

Reply:
fwiw i think we could change "`Use 'flox show <package>' to see available versions`" to "additional information" when show has more to .. show eventually making this less  duplicative